sn_networking/src/event.rs

```diff
@@ -1245,6 +1263,32 @@ impl SwarmDriver {
 
         false
     }
+
+    // if target bucket is full, remove a bootstrap node if presents.
+    fn remove_bootstrap_from_full(&mut self, peer_id: PeerId) {
```

Comment:
Ahhh, how will this work when we have many many bootstrap peers via the `network-contacts` 🤔 They'd all be removed here. Which is perhaps not ideal? (or maybe it _is_ keeping them lightweight?? )

Reply:
Perhaps we only proceed here with only 1 bootstrap node, so that is more clearly genesis. Otherwise if we provide a big list of bootstrap peers, that could be problematic?